packages/js/src/cache/notifications-cache.ts

```diff
@@ -0,0 +1,253 @@
+    newNotifications.splice(index, 1);
+
+    this.#cache.set(key, {
+      ...notificationsResponse,
+      notifications: newNotifications,
+    });
+
+    return true;
+  };
+
+  private handleNotificationEvent =
+    ({ remove }: { remove: boolean } = { remove: false }) =>
+    ({ data }: { data?: Notification }): void => {
+      if (!data) {
+        return;
+      }
+
+      const uniqueFilterKeys = new Set<string>();
+      this.#cache.keys().forEach((key) => {
+        let isNotificationFound = false;
+        if (remove) {
+          isNotificationFound = this.removeNotificationInCache(key, data);
+        } else {
+          isNotificationFound = this.updateNotificationInCache(key, data);
+        }
+
+        if (isNotificationFound) {
+          uniqueFilterKeys.add(getFilterKey(getFilter(key)));
+        }
+      });
+
+      uniqueFilterKeys.forEach((key) => {
+        const notificationsResponse = this.getAggregated(getFilter(key));
+        this.#emitter.emit('notifications.list.updated', {
+          data: notificationsResponse,
+        });
+      });
+    };
+
+  private handleNotificationsEvent =
+    ({ remove }: { remove: boolean } = { remove: false }) =>
+    ({ data }: { data?: Notification[] }): void => {
+      if (!data) {
+        return;
+      }
+
+      const uniqueFilterKeys = new Set<string>();
+      this.#cache.keys().forEach((key) => {
+        data.forEach((notification) => {
+          let isNotificationFound = false;
+          if (remove) {
+            isNotificationFound = this.removeNotificationInCache(key, notification);
+          } else {
+            isNotificationFound = this.updateNotificationInCache(key, notification);
+          }
+
+          if (isNotificationFound) {
+            uniqueFilterKeys.add(getFilterKey(getFilter(key)));
+          }
+        });
+      });
+
+      uniqueFilterKeys.forEach((key) => {
+        const notificationsResponse = this.getAggregated(getFilter(key));
+        this.#emitter.emit('notifications.list.updated', {
+          data: notificationsResponse,
+        });
+      });
+    };
+
+  private has(args: ListNotificationsArgs): boolean {
+    return this.#cache.get(getCacheKey(args)) !== undefined;
+  }
+
+  private getAggregated(filter: NotificationFilter): ListNotificationsResponse {
+    const cacheKeys = this.#cache.keys().filter((key) => {
+      const parsedFilter = getFilter(key);
+
+      return isSameFilter(parsedFilter, filter);
+    });
+
+    return cacheKeys
+      .map((key) => this.#cache.get(key))
+      .reduce<ListNotificationsResponse>(
+        (acc, el) => {
+          if (!el) {
+            return acc;
+          }
+
+          return {
+            hasMore: el.hasMore,
+            filter: el.filter,
+            notifications: [...acc.notifications, ...el.notifications],
+          };
+        },
+        { hasMore: false, filter: {}, notifications: [] }
+      );
+  }
```

Comment:
This function aggregates all the paginated notifications for the specified tab/filter and returns the combined data with the last paginated response information.